### PR TITLE
Configure with gradle

### PR DESCRIPTION
**Pull request by blundell**
Gives the option of using Gradle instead of Maven.
Adds the Gradle Wrapper to the repo, this means a user would not need to have Maven or Gradle installed its all now inside the repo.

`mvn clean compile exec:java`

is equivalent to

`gradlew clean build run`

Works exactly the same way, updated the readme to reflect this.

`build.gradle` is the main change.

`gradlew` & `gradlew.bat` are autogenerated files.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,5 @@ AwsCredentials.properties
 .idea
 target/
 *.iml
+.gradle
+build/
```
Comment (ataulm) [inline .gitignore]: add blank line to EOF?

**Comment by ataulm** [comment]
LGTM :shipit:

**Comment by blundell** [comment]
bumping a year later :-)

**Comment by xrigau** [comment]
after one year it still looks solid 💃

**Comment by Mecharyry** [comment]
@zhangzhx will you or one of the admins be able to review this? Do you need some form of legal confirmation from @blundell as per PR #12 on the README?

**Comment by zhangzhx** [comment]
@blundell 

Thanks for the contribution and sorry for the late reply. This repo is lack of maintenance and outdated. The latest AWS Java SDK version is [1.11.157](https://github.com/aws/aws-sdk-java), and we have changed the recommended coding style for using AWS Java SDK quite a bit. But updating the SDK version and coding style could be in a separate PR.

Curiously, how are you using this sample repo? We are willing to add more samples other than S3 to the repo and would like to know what is the use case from your perspective of view.

@Mecharyry yes. As this is quite a big change adding support of Gradle, we do need a verbal agreement to the license.

@blundell This PR LGTM, are you submitting this change under Apache 2.0 License?

**Comment by blundell** [comment]
Apache 2.0, yes!

**Comment by zhangzhx** [comment]
I have merged this PR. Please let us know if you have any comments or feedback to this repo.

**Pull request merged**